src/deployment.js

```diff
@@ -33,50 +45,35 @@ class Deployment {
   // by creating a deployment with that artifact
   async create(idToken) {
     try {
-      core.info(`Actor: ${this.buildActor}`)
-      core.info(`Action ID: ${this.actionsId}`)
-      const pagesDeployEndpoint = `${this.githubApiUrl}/repos/${this.repositoryNwo}/pages/deployment`
-      const artifactExgUrl = `${this.runTimeUrl}_apis/pipelines/workflows/${this.workflowRun}/artifacts?api-version=6.0-preview`
-      core.info(`Artifact URL: ${artifactExgUrl}`)
-      const { data } = await axios.get(artifactExgUrl, {
-        headers: {
-          Authorization: `Bearer ${this.runTimeToken}`,
-          'Content-Type': 'application/json'
-        }
+      core.debug(`Actor: ${this.buildActor}`)
+      core.debug(`Action ID: ${this.actionsId}`)
+      core.debug(`Actions Workflow Run ID: ${this.workflowRun}`)
+
+      const artifactUrl = await getSignedArtifactUrl({
+        runtimeToken: this.runTimeToken,
+        workflowRunId: this.workflowRun,
+        artifactName: this.artifactName
       })
-      core.info(JSON.stringify(data))
-      const artifactRawUrl = data?.value?.find(artifact => artifact.name === this.artifactName)?.url
-      if (!artifactRawUrl) {
-        throw new Error(
-          'No uploaded artifact was found! Please check if there are any errors at build step, or uploaded artifact name is correct.'
-        )
-      }
 
-      const artifactUrl = `${artifactRawUrl}&%24expand=SignedContent`
-      const payload = {
-        artifact_url: artifactUrl,
-        pages_build_version: this.buildVersion,
-        oidc_token: idToken
-      }
-      if (this.isPreview === true) {
-        payload.preview = true
-      }
-      core.info(`Creating deployment with payload:\n${JSON.stringify(payload, null, '\t')}`)
-      const response = await axios.post(pagesDeployEndpoint, payload, {
-        headers: {
-          Accept: 'application/vnd.github.v3+json',
-          Authorization: `Bearer ${this.githubToken}`,
-          'Content-type': 'application/json'
-        }
+      const deployment = await createPagesDeployment({
+        githubToken: this.githubToken,
+        artifactUrl,
+        buildVersion: this.buildVersion,
+        idToken,
+        isPreview: this.isPreview
       })
-      this.requestedDeployment = true
-      core.info(`Created deployment for ${this.buildVersion}`)
-      if (response && response.data) {
-        core.info(JSON.stringify(response.data))
-        this.deploymentInfo = response.data
+      this.deploymentInfo = {
+        ...deployment,
+        id: deployment?.id || deployment?.['status_url']?.split('/')?.pop() || this.buildVersion,
+        pending: true
       }
+
+      core.info(`Created deployment for ${this.buildVersion}, ID: ${this.deploymentInfo.id}`)
+      core.debug(JSON.stringify(deployment))
+
+      return deployment
     } catch (error) {
-      core.info(error.stack)
+      core.error(error.stack)
```

Comment:
does `core.error` show up only on failed builds? I can't remember, but I feel like there was a reason we used `.info` on this catch block.

Reply:
The only one I was aware of that gets hidden by default was `core.debug` but I'll check with the Actions team on this one. Thanks for the call out!

Reply:
Verified:

> `core.error` should show up on all builds regardless of the job failing or succeeding. And it's only an annotation, it doesn't actually set the conclusion of the job to failed, that's what `core.setFailed` is for: https://github.com/actions/toolkit/tree/main/packages/core#logging